nmostesting/NMOSTesting.py

```diff
@@ -592,60 +598,78 @@ def parse_arguments():
 
     return parser.parse_args()
 
 
-def validate_args(args):
+def validate_args(args, access_type="cli"):
+    """Validate input arguments. access_type is 'cli' for command line tool and 'http' for api use"""
+    msg = ""
+    return_type = ExitCodes.OK
     if args.list_suites:
         for test_suite in sorted(TEST_DEFINITIONS):
-            print(test_suite)
-        sys.exit(ExitCodes.OK)
+            msg += test_suite + '\n'
     elif args.describe_suites:
         for test_suite in sorted(TEST_DEFINITIONS):
-            print(test_suite + ": " + TEST_DEFINITIONS[test_suite]["name"])
-        sys.exit(ExitCodes.OK)
-
-    if "suite" in vars(args):
-        if args.suite not in TEST_DEFINITIONS:
-            print(" * ERROR: The requested test suite '{}' does not exist".format(args.suite))
-            sys.exit(ExitCodes.ERROR)
+            msg += test_suite + ": " + TEST_DEFINITIONS[test_suite]["name"] + '\n'
+    elif "suite" in vars(args):
         if args.list_tests:
             tests = enumerate_tests(TEST_DEFINITIONS[args.suite]["class"])
             for test_name in tests:
-                print(test_name)
-            sys.exit(ExitCodes.OK)
-        if args.describe_tests:
+                msg += test_name + '\n'
+        elif args.describe_tests:
             tests = enumerate_tests(TEST_DEFINITIONS[args.suite]["class"], describe=True)
             for test_description in tests:
-                print(test_description)
-            sys.exit(ExitCodes.OK)
-        if args.selection and args.selection not in enumerate_tests(TEST_DEFINITIONS[args.suite]["class"]):
-            print(" * ERROR: Test with name '{}' does not exist in test definition '{}'"
-                  .format(args.selection, args.suite))
-            sys.exit(ExitCodes.ERROR)
-        if len(args.host) != len(args.port) or len(args.host) != len(args.version):
-            print(" * ERROR: Hostnames/IPs, ports and versions must contain the same number of elements")
-            sys.exit(ExitCodes.ERROR)
-        if len(args.host) != len(TEST_DEFINITIONS[args.suite]["specs"]):
-            print(" * ERROR: This test definition expects {} Hostnames/IP(s), port(s) and version(s)"
-                  .format(len(TEST_DEFINITIONS[args.suite]["specs"])))
-            sys.exit(ExitCodes.ERROR)
-        if args.output and not args.output.endswith("xml") and not args.output.endswith("json"):
-            print(" * ERROR: Output file must end with '.xml' or '.json'")
-            sys.exit(ExitCodes.ERROR)
+                msg += test_description + '\n'
+        elif args.suite not in TEST_DEFINITIONS:
+            msg = " * ERROR: The requested test suite '{}' does not exist".format(args.suite)
+            return_type = ExitCodes.ERROR
+        elif getattr(args, "selection", "all") not in enumerate_tests(TEST_DEFINITIONS[args.suite]["class"]):
+            msg = " * ERROR: Test with name '{}' does not exist in test definition '{}'".format(

```

Comment:
I think we call it a "test suite" rather than a "test definition" from the user's point of view?

Reply:
Ha, now I look at the diff, I see most of those messages were there already, sorry!